runners/core-java/src/test/java/org/apache/beam/runners/core/ReduceFnRunnerTest.java

```diff
@@ -246,6 +247,24 @@ public void testOnElementBufferingAccumulating() throws Exception {
     tester.assertHasOnlyGlobalAndFinishedSetsFor(firstWindow);
   }
 
+  @Test
+  public void testTimerGrownSession() throws Exception {
+    ReduceFnTester<Integer, Iterable<Integer>, IntervalWindow> tester =
+        ReduceFnTester.nonCombining(
+            Sessions.withGapDuration(Duration.millis(10)),
+            DefaultTriggerStateMachine.<IntervalWindow>of(),
+            AccumulationMode.ACCUMULATING_FIRED_PANES,
+            Duration.millis(50),
+            ClosingBehavior.FIRE_ALWAYS);
+
+    tester.setAutoAdvanceOutputWatermark(true);
+
+    tester.advanceInputWatermark(new Instant(0));
+    injectElement(tester, 1);
+
+    tester.advanceInputWatermark(new Instant(100));
```

Comment:
Should indicate what this tests -- specifically, what happens if the input watermark jumps from before EOW to after GC time.